data/amazon-simpleDB/src/main/java/com/teiid/spring/data/amazon/simpleDB/SimpleDBConfiguration.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teiid.spring.data.amazon.simpleDB;
+
+import org.teiid.translator.simpledb.api.BaseSimpleDBConfiguration;
+
+public class SimpleDBConfiguration implements BaseSimpleDBConfiguration {
+    private String accessKey;
+    private String secretAccessKey;
```

Comment:
`secretAccessKey` is password? this variable seems odd? How it called on SimpleDB?

Reply:
It's actually secretKey of aws.
Originially in the wildfly module, this was the variable name throughtout, that's why this. 

https://github.com/ciphereck/teiid/blob/d3737491eae3ad49b9a6d41c8452d0e969cf092a/wildfly/connectors/amazon/connector-simpledb/src/main/java/org/teiid/resource/adapter/simpledb/SimpleDBConnectionImpl.java#L48

I didn't understood "How it called on SimpleDB?" part?

Reply:
check with amazon-athena, or s3 nomenclature for properties see if you can map to that in spring-boot

Reply:
Well in s3, it's secretKey only.
But to do mapping, we need to change interface in teiid-parent too.

Since, in last SimpleDBConnectionImpl, it was secretAccessKey, I took the same as forward.

Reply:
I'm fine with changing this on the teiid side to just secretKey.  We can leave the resource adapter with the secretAcessKey for backwards compatibility.

Reply:
I didn't get it. Here when you say to teiid side, do you mean teiid-parent or teiid-spring-boot? Cause you want me to leave resource adapter (I think it means connector in teiid-parent).

Anyways this is what I came up with: [ciphereck@652a44](https://github.com/ciphereck/teiid/commit/652a449)
I'm not sure, whether changing this, will trigger any issue or not. If this seems like a good idea, I can create a PR for the same. Otherwise, please let me know how to proceed.

Reply:
`teiid-parent` project is the main `teiid` project, and then `teiid-spring-boot` is dependent on it. but both are separate projects, teiid-spring-boot depends on teiid but not other way around, so when you write code you need to follow that.

Reply:
Yeah. But to change the getter from getSecretAccessKey() to getSecretKey(), we need to change interface it in the teiid-parent simpledb module.

Reply:
yes, go ahead and change there and submit a PR, but as Steve says do not change in the resource-adapter part as it will break the backward compatibility. so resource-adapter's SimpleDbConfiguration class you will have the new method and old but old method will be used as set/get and new method just uses values from old one.

Reply:
If we use old method as set/get only, will it situation, we are facing currently?
What I could understand is we now have two method, old one which return secretAccessKey and a new default method in interface which return value from old one. I think this is the only change we need to incorporate it in spring-boot.

Am I right here?

Reply:
teiid/teiidt#1311

Reply:
In spring-boot ignore the old method and only define getters and setters for new one. To support the interface you can have the getter for the old but pointing to the new method